internal/grpc/services/ocmshareprovider/ocmshareprovider.go

```diff
@@ -144,7 +149,31 @@ func (s *service) CreateOCMShare(ctx context.Context, req *ocm.CreateOCMShareReq
 		}, nil
 	}
 
-	share, err := s.sm.Share(ctx, req.ResourceId, req.Grant, name, req.RecipientMeshProvider, permissions, nil, "", ocm.Share_SHARE_TYPE_REGULAR)
+	// discover share type
+	sharetype := ocm.Share_SHARE_TYPE_REGULAR
```

Comment:
This hack is wrong. The share type should be specified in all locations where `CreateOCMShare` is being called.

Reply:
When a different share type is discovered (eg. 'datatx') the `sharetype` is set to represent that:
```
			if sp.Name == "datatx" {
				sharetype = ocm.Share_SHARE_TYPE_TRANSFER
			}
```